tests/acceptance/01_vars/02_functions/isconnectable.cf

```diff
@@ -0,0 +1,28 @@
+body common control
+{
+      inputs => { "../../default.cf.sub" };
+      bundlesequence  => { default("$(this.promise_filename)") };
+      version => "1.0";
+}
+
+bundle agent check
+{
+  meta:
+    "test_skip_unsupported" string => "!linux";
+
+  vars:
+    # to avoid dependencies (like installing netstat/net-utils), look at proc filesystem for listening port, probably at least one (like ssh or chrony)
+    # header of /proc/net/tcp is: sl  local_address rem_address   st tx_queue rx_queue tr tm->when retrnsmt   uid  timeout inode
+    # local_address is ip:port
+    # example line for ssh (port 22, hex 16): 0: 00000000:0016 00000000:0000 0A 00000000:00000000 00:00000000 00000000     0        0 25575 2 ffff9eac03038000 100 0 0 10 0
+    "listening_port" string => execresult("cat /proc/net/tcp | head -2 | tail -1 | awk '{print $2}' | cut -d: -f2 | xargs -I{} printf '%d\n' 0x{}", "useshell");
```

Comment:
```suggestion
    "listening_port" string => execresult("awk 'NR == 2 {print $2}' /proc/net/tcp | cut -d: -f2 | xargs -I{} printf '%d\n' 0x{}", "useshell");
```

A lil bit shorter

Reply:
Appreciate it but I don't know awk that well yet. Bottomless pit this posix scripting is! I like mine better because it is less obfuscated. I can guess that NR == 2 means "take record number 2, aka 2nd line" but :shrug: